mypyc/test-data/irbuild-basic.test

```diff
@@ -3546,3 +3546,275 @@ L0:
     r2 = PyObject_Vectorcall(r1, 0, 0, 0)
     r3 = box(None, 1)
     return r3
+
+[case testForFilterBool]
+def f(x: int) -> bool:
+    return bool(x % 2)
+def g(a: list[int]) -> int:
+    s = 0
+    for x in filter(f, a):
+        s += x
+    return s
+[out]
+def f(x):
+    x, r0 :: int
+    r1 :: bit
+L0:
+    r0 = CPyTagged_Remainder(x, 4)
+    r1 = r0 != 0
+    return r1
+def g(a):
+    a :: list
+    s :: int
+    r0 :: dict
+    r1 :: str
+    r2 :: object
+    r3, r4 :: native_int
+    r5 :: bit
+    r6 :: object
+    r7, x :: int
+    r8 :: bool
+    r9 :: int
+    r10 :: native_int
+L0:
+    s = 0
+    r0 = __main__.globals :: static
+    r1 = 'f'
+    r2 = CPyDict_GetItem(r0, r1)
+    r3 = 0
+L1:
+    r4 = var_object_size a
+    r5 = r3 < r4 :: signed
+    if r5 goto L2 else goto L6 :: bool
+L2:
+    r6 = list_get_item_unsafe a, r3
+    r7 = unbox(int, r6)
+    x = r7
+    r8 = f(x)
+    if r8 goto L4 else goto L3 :: bool
+L3:
+    goto L5
+L4:
+    r9 = CPyTagged_Add(s, x)
+    s = r9
+L5:
+    r10 = r3 + 1
+    r3 = r10
+    goto L1
+L6:
+L7:
+    return s
+
+[case testForFilterInt]
+def f(x: int) -> int:
+    return x % 2
+def g(a: list[int]) -> int:
+    s = 0
+    for x in filter(f, a):
+        s += x
+    return s
+[out]
+def f(x):
+    x, r0 :: int
+L0:
+    r0 = CPyTagged_Remainder(x, 4)
+    return r0
+def g(a):
+    a :: list
+    s :: int
+    r0 :: dict
+    r1 :: str
+    r2 :: object
+    r3, r4 :: native_int
+    r5 :: bit
+    r6 :: object
+    r7, x, r8 :: int
+    r9 :: bit
+    r10 :: int
+    r11 :: native_int
+L0:
+    s = 0
+    r0 = __main__.globals :: static
+    r1 = 'f'
+    r2 = CPyDict_GetItem(r0, r1)
+    r3 = 0
+L1:
+    r4 = var_object_size a
+    r5 = r3 < r4 :: signed
+    if r5 goto L2 else goto L6 :: bool
+L2:
+    r6 = list_get_item_unsafe a, r3
+    r7 = unbox(int, r6)
+    x = r7
+    r8 = f(x)
+    r9 = r8 != 0
+    if r9 goto L4 else goto L3 :: bool
+L3:
+    goto L5
+L4:
+    r10 = CPyTagged_Add(s, x)
+    s = r10
+L5:
+    r11 = r3 + 1
+    r3 = r11
+    goto L1
+L6:
+L7:
+    return s
+
+[case testForFilterStr]
```

Comment:
This seems specialized enough to move to a run test (see my comments in other PRs about our conventions related to irbuild tests).